app/src/main/java/com/github/se/gomeet/ui/mainscreens/EventInfo.kt

```diff
@@ -0,0 +1,308 @@
+import com.google.android.gms.maps.model.LatLng
+import com.google.maps.android.compose.GoogleMap
+import com.google.maps.android.compose.Marker
+import com.google.maps.android.compose.rememberCameraPositionState
+import com.google.maps.android.compose.rememberMarkerState
+
+@Composable
+fun EventInfo(
+    nav: NavigationActions,
+    title: String = "",
+    date: String = "",
+    time: String = "",
+    organizer: String = "",
+    rating: Double = 0.0,
+    image: Painter = painterResource(id = R.drawable.chess_demo),
+    description: String = "",
+    loc: LatLng = LatLng(0.0, 0.0)
+) {
+  Log.d("EventInfo", "Loc is $loc")
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            backgroundColor = MaterialTheme.colorScheme.background,
+            elevation = 0.dp,
+            title = {
+              // Empty title since we're placing our own components
+            },
+            navigationIcon = {
+              IconButton(onClick = { nav.goBack() }) {
+                Icon(
+                    imageVector = Icons.AutoMirrored.Filled.ArrowBack,
+                    contentDescription = "Back",
+                    tint = MaterialTheme.colorScheme.onBackground)
+              }
+            },
+            actions = {
+              IconButton(onClick = { /* Handle share action */}) {
+                Icon(
+                    imageVector =
+                        ImageVector.vectorResource(id = R.drawable.baseline_chat_bubble_outline_24),
+                    contentDescription = "Share",
+                    modifier = Modifier.size(24.dp),
+                    tint = MaterialTheme.colorScheme.onBackground)
+              }
+
+              IconButton(onClick = { /* Handle more action */}) {
+                Icon(
+                    imageVector = Icons.Filled.MoreVert,
+                    contentDescription = "More",
+                    modifier = Modifier.rotate(90f),
+                    tint = MaterialTheme.colorScheme.onBackground)
+              }
+            })
+      },
+      bottomBar = {
+        // Your bottom bar content
+      }) { innerPadding ->
+        Column(
+            modifier =
+                Modifier.padding(innerPadding)
+                    .padding(start = 15.dp, end = 15.dp, top = 0.dp, bottom = 15.dp)
+                    .fillMaxSize()
+                    .verticalScroll(state = rememberScrollState())) {
+              EventHeader(
+                  title = title,
+                  organizer = organizer,
+                  rating = rating,
+                  nav = nav,
+                  date = date,
+                  time = time)
+              Spacer(modifier = Modifier.height(20.dp))
+              EventButtons()
+              Spacer(modifier = Modifier.height(20.dp))
+              EventImage(painter = image)
+              Spacer(modifier = Modifier.height(20.dp))
+              EventDescription(text = description)
+              Spacer(modifier = Modifier.height(20.dp))
```

Comment:
Using hardcoded values for spacer heights may cause layout issues on smaller screens. Have you had a chance to test how these spacers scale across different device sizes? It might be beneficial to consider a responsive approach consistency on all devices.

Reply:
The screens is scrollable so vertical spacers aren't a problem even on smaller devices.